examples/example-hono-api/src/auth0.ts

```diff
@@ -0,0 +1,166 @@
+                audience: auth0_audience,
+            });
+        }
+
+        const credentials = ctx.req.raw.headers.get('Authorization');
+        if (!credentials) {
+            const errDescription = 'No Authorization header included in request';
+            throw new HTTPException(401, {
+                message: errDescription,
+                res: unauthorizedResponse({
+                    ctx,
+                    error: 'invalid_request',
+                    errDescription,
+                }),
+            });
+        }
+
+        const parts = credentials.split(/\s+/);
+        if (parts.length !== 2) {
+            const errDescription = 'Invalid Authorization header structure';
+            throw new HTTPException(401, {
+                message: errDescription,
+                res: unauthorizedResponse({
+                    ctx,
+                    error: 'invalid_request',
+                    errDescription,
+                }),
+            });
+        }
+
+        if (parts[0] !== 'Bearer') {
+            const errDescription =
+                'Invalid authorization header (only Bearer tokens are supported)';
+            throw new HTTPException(401, {
+                message: errDescription,
+                res: unauthorizedResponse({
+                    ctx,
+                    error: 'invalid_request',
+                    errDescription: errDescription,
+                }),
+            });
+        }
+
+        const accessToken = parts[1];
+        if (!accessToken || accessToken.length === 0) {
+            const errDescription = 'No token included in request';
+            throw new HTTPException(401, {
+                message: errDescription,
+                res: unauthorizedResponse({
+                    ctx,
+                    error: 'invalid_request',
+                    errDescription,
+                }),
+            });
+        }
+
+        try {
+            const jwtPayload = await apiClient.verifyAccessToken({accessToken});
+
+            ctx.set('user', jwtPayload);
+
+            await next();
+        } catch (cause) {
+            console.trace(cause);
+
+            throw new HTTPException(401, {
+                message: 'Unauthorized',
+                res: unauthorizedResponse({
+                    ctx,
+                    error: 'invalid_token',
+                    statusText: 'Unauthorized',
+                    errDescription: 'Token verification failure',
+                }),
+                cause,
+            });
+        }
+
+    };
+};
+
+function unauthorizedResponse(opts: {
+    ctx: Context;
+    error: string;
+    errDescription: string;
+    statusText?: string;
+}) {
+    return new Response('Unauthorized', {
+        status: 401,
+        statusText: opts.statusText,
+        headers: {
+            'WWW-Authenticate': `Bearer realm="${opts.ctx.req.url}",error="${opts.error}",error_description="${opts.errDescription}"`,
+        },
+    });
+}
```

Comment:
```suggestion
function unauthorizedResponse(opts: {
    ctx: Context;
    error: string;
    errorDescription: string;
    statusText?: string;
}) {
    return new Response('Unauthorized', {
        status: 401,
        statusText: opts.statusText,
        headers: {
            'WWW-Authenticate': `Bearer realm="${opts.ctx.req.url}",error="${opts.error}",error_description="${opts.errorDescription}"`,
        },
    });
}
```

I think using `err` over `error` isn't helpful here, even mopre so because line 123 already mentions `error`.